routing/router_test.go

```diff
@@ -2820,4 +2821,21 @@ func TestFindBlindedPathsWithMC(t *testing.T) {
 		"alice,bob,dave",
 		"alice,frank,dave",
 	})
+
+	// Test that if the user explicitly indicates that we should ignore
+	// the Frank node during path selection, then this is done.
+	routes, err = ctx.router.FindBlindedPaths(
+		dave, 1000, probabilitySrc, &BlindedPathRestrictions{
+			MinDistanceFromIntroNode: 2,
+			NumHops:                  2,
+			MaxNumPaths:              3,
+			NodeOmissionSet: fn.NewSet[route.Vertex](
+				frank,
+			),
+		},
+	)
+	require.NoError(t, err)
+	assertPaths(routes, []string{
+		"alice,frank,dave",
```

Comment:
hm, this should exclude `frank`, right?

Reply:
great catch! there was a bug in the test assertion and in the probability sorting function